### PR TITLE
Reorg installfest

### PR DESCRIPTION
**Pull request by decklin**
Paging @janetriley - sorry for delay! I started this after the last workshop but wanted to clean it up.

This is a big reorganization, and will need editing after it lands - I will try to find as many OSes as possible to test on, and make the per-OS parts more specific.

Main change is the new downloads page, which uses a table. Students find the column for their OS, and download each file in the column. I got positive feedback on this in a real-world "hey, look at this" test. Hopefully it feels more streamlined to match the streamlined purpose (just download the files). Made the "re-download in case versions changed" bigger, because they'll always change (I'll update versions after this lands).

The installfest page splits out the per-OS instructions into separate pages -- we sort of did that before, but some VM instructions were mixed in with some OS instructions. Goal here is to only cover how you run an installer on the particular OS, and then merge as much of the VM setup steps as possible into the VM setup page, which is the next step on the checklist. These steps need some tuning! For example we can be specific with filenames instead of ".exe or .msi" But this is just step 1, which is applying some organization.

(one thing there is that we may want to move back the creation of `~/workspace` to the OS pages, if we have to.)

Instructions themselves: I've tried to give the DIY Linux instructions some love, and added a new platform, Arch. They need real-world testing, though. OS X Xcode downloads are updated as well. I've added a hint that you can also do these as a sort of "expert mode", and if any students show interest, I could lead a breakout session on how to install "real" Ruby after you complete everything as a way to refine those steps.

One final thing: to do tables (like with alert boxes), you have to drop down to HTML. This led me away from doing anything fancy where prose was in the table cells. However, if in the future someone wants to add a JS library that makes the per-OS instructions into a tab sheet, that would be great!

Any other suggestions (from anyone out there) are very welcome. I hope that this is a step toward making future tweaks (which tend to be OS-specific) easier for volunteers to do and us to merge. I am busy until Friday but can follow up with feedback after that.

### CODE REVIEW

**Comment by danchoi** [comment]
You rock!

On Tue, Jul 29, 2014 at 07:51 PM, Decklin Foster notifications@github.com wrote:

> from: Decklin Foster notifications@github.com
> date: Tue, Jul 29 04:51 PM -07:00 2014
> to: railsbridge-boston/railsbridge-boston railsbridge-boston@noreply.github.com
> reply-to: railsbridge-boston/railsbridge-boston reply@reply.github.com
> subject: [railsbridge-boston] Reorg installfest (#158)
> 
> Paging @janetriley - sorry for delay! I started this after the last workshop but wanted to clean it up.
> 
> This is a big reorganization, and will need editing after it lands - I will try to find as many OSes as possible to test on, and make the per-OS parts more specific.
> 
> Main change is the new downloads page, which uses a table. Students find the column for their OS, and download each file in the column. I got positive feedback on this in a real-world &quot;hey, look at this&quot; test. Hopefully it feels more streamlined to match the streamlined purpose (just download the files). Made the &quot;re-download in case versions changed&quot; bigger, because they&#39;ll always change (I&#39;ll update versions after this lands).
> 
> The installfest page splits out the per-OS instructions into separate pages -- we sort of did that before, but some VM instructions were mixed in with some OS instructions. Goal here is to only cover how you run an installer on the particular OS, and then merge as much of the VM setup steps as possible into the VM setup page, which is the next step on the checklist. These steps need some tuning! For example we can be specific with filenames instead of &quot;.exe or .msi&quot; But this is just step 1, which is applying some organization.
> 
> (one thing there is that we may want to move back the creation of `~/workspace` to the OS pages, if we have to.)
> 
> Instructions themselves: I&#39;ve tried to give the DIY Linux instructions some love, and added a new platform, Arch. They need real-world testing, though. OS X Xcode downloads are updated as well. I&#39;ve added a hint that you can also do these as a sort of &quot;expert mode&quot;, and if any students show interest, I could lead a breakout session on how to install &quot;real&quot; Ruby after you complete everything as a way to refine those steps.
> 
> One final thing: to do tables (like with alert boxes), you have to drop down to HTML. This led me away from doing anything fancy where prose was in the table cells. However, if in the future someone wants to add a JS library that makes the per-OS instructions into a tab sheet, that would be great!
> 
> Any other suggestions (from anyone out there) are very welcome. I hope that this is a step toward making future tweaks (which tend to be OS-specific) easier for volunteers to do and us to merge. I am busy until Friday but can follow up with feedback after that.
> You can merge this Pull Request by running:
> 
>   git pull https://github.com/railsbridge-boston/railsbridge-boston reorg-installfest
> 
> Or you can view, comment on it, or merge it online at:
> 
>   https://github.com/railsbridge-boston/railsbridge-boston/pull/158
> 
> -- Commit Summary --
> - Remove orphaned console2 page
> - Reorganize download/installfest, downloads table, new linux VM, mac/linux non-VM instructions
> - Split per-OS installer instructions back into separate pages
> - Add links back to downloads page
> 
> -- File Changes --
> 
> ```
> M public/stylesheets/site.css (12)
> A views/downloads.markdown (88)
> A views/downloads/linux.markdown (49)
> A views/downloads/xcode.markdown (26)
> D views/friday.markdown (55)
> A views/installfest.markdown (86)
> M views/installfest/checklist.markdown (2)
> A views/installfest/command_line.markdown (32)
> R views/installfest/command_line_basics.markdown (31)
> D views/installfest/install/linux.markdown (99)
> D views/installfest/install/osx.markdown (9)
> D views/installfest/install/windows.markdown (49)
> D views/installfest/install/windows_console2.markdown (115)
> A views/installfest/linux.markdown (22)
> A views/installfest/linux_novm.markdown (77)
> A views/installfest/osx.markdown (25)
> R views/installfest/osx_novm.markdown (16)
> M views/installfest/text_editor.markdown (47)
> M views/installfest/vm_setup.markdown (71)
> A views/installfest/windows_7.markdown (26)
> A views/installfest/windows_xp.markdown (39)
> M views/navbar.erb (6)
> D views/pre_workshop.markdown (58)
> M web.rb (9)
> ```
> 
> -- Patch Links --
> 
> https://github.com/railsbridge-boston/railsbridge-boston/pull/158.patch
> https://github.com/railsbridge-boston/railsbridge-boston/pull/158.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/railsbridge-boston/railsbridge-boston/pull/158

---

Sent from Vmail
http://danielchoi.com/software/vmail.html

File: views/downloads/xcode.markdown [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+click Register. Fill out the first page with your account information.
+The second page is a developer survey, which you can skip. Just go to
+the bottom and click "Continue".
+
+Confirm your account and log in.
+
+# OS X 10.5
+
+Download
+[Xcode 3.1.4](http://adcdownload.apple.com/Developer_Tools/xcode_3.1.4_developer_tools/xcode314_2809_developerdvd.dmg)
```
Comment (mxie) [inline views/downloads/xcode.markdown]: I get an Access Denied error when I try to visit this page:

access denied 2014-08-12 11-42-03 2014-08-12 11-42-11

File: views/installfest/command_line_basics.markdown [hunk 2]
```diff
@@ -1,4 +1,6 @@
-# Using the Command Line
+# Command Line Tutorial
+
+<!-- TODO: this should be rewritten before the next Intro workshop. -->
```
Comment (mxie) [inline views/installfest/command_line_basics.markdown:3]: Would this portion from the Intro workshop work at all?: http://docs.railsbridgeboston.org/ruby/command_line. Maybe we should bring that content over here.

**Comment by mxie** [comment]
Just left a couple of comments, but this is _awesome_ work @decklin! Thank you so much for putting so much time into this!

File: views/downloads/xcode.markdown (hunk 1 above)
Comment (decklin) [inline views/downloads/xcode.markdown]: I vaguely remember checking this file while I was logged in (it's a really out of date version, but they did have at least something available)... I think you have to register and log in to be able to see the list or even download the dmg. I will check again when I have some time.

(This is only for people who need to download Xcode onto a pre-App-Store OS X version, at least)

Comment (decklin) [inline views/downloads/xcode.markdown]: Pushed an update to this link that should take you to an Apple ID login page if you're not logged in yet.

File: views/installfest/command_line_basics.markdown (hunk 2 above)
Comment (decklin) [inline views/installfest/command_line_basics.markdown:3]: Yeah, I want to take a good look at that once this page is moved.

**Comment by mxie** [comment]
This looks great! Feel free to merge this to master when you're ready.

**Comment by decklin** [comment]
OK, merging now! My next step will be to check for updated downloads and build a new VM (this week or next), then I'll try following the instructions on as many platforms as I can find.

**Comment by mxie** [comment]
Awesome - sounds good! Do you need any help with trying out the different platforms? We might be able to grab some volunteers.

**Comment by decklin** [comment]
That would be great - could we have people try with the 2014-09-rc1 box in https://github.com/railsbridge-boston/railsbridge-boston/pull/159 ? Thanks!

**Pull request merged**